Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,199 @@
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnGuardianCreated(EntityUid uid, GuardianCreatorComponent component, UseInHandEvent args)
+        {
+            //Only works if the guardian isn't already present. NOTE: this is up to preference, we can probably later allow
+            //a person to host multiple guardians, however the idea is that in a normal setting you won't get the opportunity to
+            if (!args.User.TryGetComponent<GuardianHostComponent>(out var guardianComponent))
+            {
+                if (args.User.TryGetComponent<SharedActionsComponent>(out SharedActionsComponent? action))
+                {
+                    if (component.Used == false)
+                    {
+                        var hostcomp = args.User.AddComponent<GuardianHostComponent>();
+                        var guardian = EntityManager.SpawnEntity(component.GuardianType, hostcomp.Owner.Transform.Coordinates);
+                        hostcomp.GuardianContainer.Insert(guardian);
+                        hostcomp._hostedguardian = guardian.Uid;
+                        //Ensures the guardian component on the supposed guardian entity and fills out it's synced parametres
+                        var guardiancomp = guardian.EnsureComponent<GuardianComponent>();
+                        guardiancomp.Guardian = guardian.Uid;
+                        guardiancomp.Host = args.User.Uid;
+                        //Grant the user the recall action and notify them
+                        action.Grant(ActionType.ManifestGuardian);
+                        args.User.PopupMessage(Loc.GetString("guardian-created"));
+                        //Exhaust the activator
+                        component.Used = true;
+                    }
+                    else
+                    {
+                        args.User.PopupMessage(Loc.GetString("guardian-activator-empty-invalid-creaton"));
+                    }
+                }
+                else
+                {
+                    args.User.PopupMessage(Loc.GetString("guardian-no-actions-invalid-creation"));
+                }
+            }
+            else
+            {
+                args.User.PopupMessage(Loc.GetString("guardian-already-present-invalid-creation"));
+            }
+        }
+
+        /// <summary>
+        /// Called every time the host moves, to make sure the distance between the host and the guardian isn't too far
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnGuardianHostMove(EntityUid uid, GuardianHostComponent component, ref MoveEvent args)
+        {
+            var guardcomp = EntityManager.GetEntity(component._hostedguardian).GetComponent<GuardianComponent>();
```

Comment:
Do not GetComponent unless you know it exists (you don't), also don't do needless GetEntity calls.

Reply:
that's why ensured it beforehand, I'll recode it to try to get it